test/src/main/java/org/sample/NaiveImplementation.java

```diff
@@ -0,0 +1,52 @@
+package org.sample;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.lang.Math;
+
+public class NaiveImplementation {
+  private double square(double num) {
+    return (num * num);
+  }
+
+  private double distanceSquared(Coordinates start, Coordinates end) {
+    return square(start.getLng() - end.getLng()) + square(start.getLat() - end.getLat());
+  }
+
+  private double distanceFromSegment(Coordinates start, Coordinates end, Coordinates point) {
+    double segmentDistanceSquared = distanceSquared(start, end);
+
+    if (segmentDistanceSquared == 0) {
+      return Math.sqrt(distanceSquared(start, point));
+    }
+
+    double projectionScale = ((point.getLng() - start.getLng()) * (end.getLng() - start.getLng())
+        + (point.getLat() - start.getLat()) * (end.getLat() - start.getLat())) / segmentDistanceSquared;
+    projectionScale = Math.max(0, Math.min(1, projectionScale));
+
+    Coordinates projection = new Coordinates(start.getLat() + projectionScale * (end.getLat() - start.getLat()),
+        start.getLng() + projectionScale * (end.getLng() - start.getLng()));
+
+    return Math.sqrt(distanceSquared(point, projection));
+  }
```

Comment:
nit: Might make more sense to put this under reference folder as a method in coordinate class?

Reply:
I was basically trying to keep the classes in the reference folder identical to the one in our actual web-app, that way if we make changes to our implementation we can just copy the new file wholesale. Naive implementation maybe wasn't a very good name? It's just the servlet code made non-servlet dependent.

Reply:
I moved all the math to a new file Distance.java in the next PR (so quadtree can use it too). The reason I'm not modifying any files in the reference folder is because I'm trying to keep it so that anyone who wants to run the test can just copy the code wholesale.